Bottom: 3d7f8cdb9b4cea1468c9dbaa0e8943b1015c3137
Top:    c5bd0b81f3e56159d325a97a21cc0d306515e9ec
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-03-22 14:19:01 +0800

Refresh of vhost_net-prefetch-desc

---

diff --git a/drivers/vhost/net.c b/drivers/vhost/net.c
index 0ef4f3f..74d5961 100644
--- a/drivers/vhost/net.c
+++ b/drivers/vhost/net.c
@@ -698,14 +698,10 @@ static void handle_rx_batched(struct vhost_net *net)
 		}
 		for (i = 0; i < avails; i++) {
 			int len =__skb_array_len_with_tag(rvq->rxq[rvq->rh + i]);
-			
-			vhost_add_used(indices[i], cpu_to_vhost32(vq, len));
+			vhost_add_used_elem(vq, indices[i],
+					    cpu_to_vhost32(vq, len), i);
 		}
-
-		/* Make sure buffer is written before we update index. */
-		smp_wmb();
-
-		
+		while (nvq->rh != nvq->rt) {
 	}
 }
